runtime/gc_vlhgc/InterRegionRememberedSet.hpp

```diff
@@ -185,14 +189,46 @@ class MM_InterRegionRememberedSet : public MM_BaseVirtual
 	void clearFromRegionReferencesForCompactOptimized(MM_EnvironmentVLHGC* env);
 	MM_InterRegionRememberedSet(MM_HeapRegionManager *heapRegionManager);
 	
+
+	/**
+	 * Helper function isCompressedCardDirtyForPartialCollect
+	 * extended from compressedCardTable->isCompressedCardDirtyForPartialCollect(), only in case first PGC after GMP, also return as dirty if card Contains No Object.
+	 */
+	MMINLINE static bool isCompressedCardDirtyForPartialCollect(MM_EnvironmentVLHGC* env, UDATA card, MM_CompressedCardTable *compressedCardTable, MM_MarkMap *markMap)
+	{
+		void* heapAddress = convertHeapAddressFromRememberedSetCard(card);
+		bool ret = compressedCardTable->isCompressedCardDirtyForPartialCollect(env, heapAddress);
+		if (!ret && (NULL != markMap)) {
+			/* TODO: consider incorporating areAnyLiveObjectsInCard info when building CompressedCard */
+			ret = !markMap->areAnyLiveObjectsInCard(heapAddress);
+		}
+		return ret;
+
+	}
 protected:
 public:
+	/**
+	 * Helper function cardMayContainObjects
+	 * if this is first PGC after GMP, check if markMap->areAnyLiveObjectsInCard()
+	 * otherwise check if region->containsObjects()
+	 */
+	MMINLINE static bool cardMayContainObjects(UDATA card, MM_HeapRegionDescriptorVLHGC *fromRegion, MM_MarkMap *markMap)
```

Comment:
These helpers won't be able to be static, afterall :( . It may work now, but once OMR_GC_FULL_POINTERS is enabled, they will indirectly rely on class member _compressObjectReferences